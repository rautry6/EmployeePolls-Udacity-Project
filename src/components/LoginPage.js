import { connect } from "react-redux/es/exports";
import { useState} from "react";
import { setAuthedUser } from "../actions/authedUser";
import { useNavigate } from "react-router-dom";

const LoginPage = (props) => {
  const nav = useNavigate();
  let [username, setUsername] = useState("");
  let [password, setPassword] = useState("");
  let [error, setError] = useState("");


  let users = props.users;

  const handleSubmit = (e) => {
    console.log(username, password);
    if (username in users) {
      if (users[username].password === password) {
        props.dispatch(setAuthedUser(users[username]));

        nav("/");
      } else {
        setUsername("");
        setPassword("");
        setError(true);
      }
    } else {
      setUsername("");
      setPassword("");
      setError(true);
    }
  };

  const handleUsername = (e) => {
    setUsername(e.target.value);
  };

  const handlePassword = (e) => {
    setPassword(e.target.value);
  };

  return (
    <div className="login-page">
      <h1>Employee Polls</h1>
      <h2>Login</h2>
      <div>
        {error ? (
          <div>
            <p itemID="invalid-login">Invalid username or password</p>{" "}
            <p>User</p>
            <input type="text" value={username} onChange={handleUsername} />
            <p>Password</p>
            <input type="password" value={password} onChange={handlePassword} />
            <p>
              <button onClick={handleSubmit}>Login</button>
            </p>
          </div>
        ) : (
          <div>
            <p>User</p>
            <input type="text" onChange={handleUsername} />
            <p>Password</p>
            <input type="password" onChange={handlePassword} />
            <p>
              <button onClick={handleSubmit}>Login</button>
            </p>{" "}
          </div>
        )}
      </div>
    </div>
  );
};

function mapStateToProps(state) {
  return {
    users: state.users,
  };
}

export default connect(mapStateToProps)(LoginPage);
