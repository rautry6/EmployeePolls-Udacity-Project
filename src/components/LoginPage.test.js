import LoginPage from "./LoginPage.js";
import { render } from "@testing-library/react";
import { createStore } from "redux";
import rootReducer from "../reducers";
import middleware from "../middleware";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

const store = createStore(rootReducer, middleware);
describe("LoginPage", () => {
  it("matches snapshot", () => {
    const { container } = render(<Provider store={store}>
    <BrowserRouter>
      <LoginPage />
    </BrowserRouter>
      </Provider>);
    expect(container).toMatchSnapshot();
  });
});
