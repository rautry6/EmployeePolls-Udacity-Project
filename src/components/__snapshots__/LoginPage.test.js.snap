// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage matches snapshot 1`] = `
<div>
  <div
    class="login-page"
  >
    <h1
      class="login-title"
    >
      Employee Polls
    </h1>
    <h2
      class="login-header"
    >
      Login
    </h2>
    <div
      class="login-field"
    >
      <div>
        <p
          class="username"
        >
          Username
        </p>
        <input
          class="login-input"
          data-testid="login-input"
          type="text"
          value=""
        />
        <p
          class="password"
        >
          Password
        </p>
        <input
          class="password-input"
          data-testid="password-input"
          type="password"
          value=""
        />
        <p>
          <button
            class="submit-button"
            data-testid="submit-button"
          >
            Login
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;
